### PR TITLE
chore: Add instance type offering availability to metrics

### LINKED ISSUE #2596: Expose Cloud Provider Offering Availability as Metrics
### Tell us about your request

This idea was originally proposed by @bwagner5.

The idea here would be to expose a gauge metric like `karpenter_cloudprovider_offering_available` with labels `instance_type`, `capacity_type`, and `zone`. 

This metric would be set to 1 for all offerings that Karpenter is aware are available from the cloud provider. If the cloudprovider changed its mind about the availability of the offering (i.e. for AWS, there is an ICE [Insufficient Capacity Error] when launching an instance) then the gauge would be exposed as 0

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

N/A

### Are you currently working around this issue?

N/A

### Additional Context

_No response_

### Attachments

_No response_

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #2596

**Description**

Adds a metric to our CloudProvider metrics to track the offering availability over time. This also updates the `karpenter_cloudprovider_instance_type_price_estimate` metric to be fired when the offerings are created, rather than when the prices are updated in the pricing provider. This also changes the name to include the word "offering" to be `karpenter_cloudprovider_instance_type_offering_price_estimate` to align with `karpenter_cloudprovider_instance_type_offering_available`

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/providers/instancetype/instancetype.go [hunk 1]
```diff
@@ -118,16 +118,15 @@ func (p *Provider) List(ctx context.Context, kc *corev1beta1.KubeletConfiguratio
...
+		instanceTypeVCPU.With(prometheus.Labels{
+			instanceTypeLabel: *i.InstanceType,
+		}).Set(float64(aws.Int64Value(i.VCpuInfo.DefaultVCpus)))
+		instanceTypeMemory.With(prometheus.Labels{
+			instanceTypeLabel: *i.InstanceType,
+		}).Set(float64(aws.Int64Value(i.MemoryInfo.SizeInMiB) * 1024 * 1024))
+
 		return NewInstanceType(ctx, i, kc, p.region, nodeClass, p.createOfferings(ctx, i, instanceTypeOfferings[aws.StringValue(i.InstanceType)], zones, subnetZones))
 	})
-	for _, instanceType := range instanceTypes {
```
Comment (jonathan-innis) [inline pkg/providers/instancetype/instancetype.go:123]: Avoids additional iteration that we don't need that can cause additional latency

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: website/content/en/preview/upgrading/upgrade-guide.md [hunk 2]
```diff
@@ -37,6 +37,14 @@ kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef
 WHEN CREATING A NEW SECTION OF THE UPGRADE GUIDANCE FOR NEWER VERSIONS, ENSURE THAT YOU COPY THE BETA API ALERT SECTION FROM THE LAST RELEASE TO PROPERLY WARN USERS OF THE RISK OF UPGRADING WITHOUT GOING TO 0.32.x FIRST
 -->
 
+### Upgrading to `0.36.0`+
+
+{{% alert title="Warning" color="warning" %}}
+`0.33.0`+ _only_ supports Karpenter v1beta1 APIs and will not work with existing Provisioner, AWSNodeTemplate or Machine alpha APIs. Do not upgrade to `0.35.0`+ without first [upgrading to `0.32.x`]({{<ref "#upgrading-to-0320" >}}). This version supports both the alpha and beta APIs, allowing you to migrate all of your existing APIs to beta APIs without experiencing downtime.
```
Comment (engedaam) [inline website/content/en/preview/upgrading/upgrade-guide.md:43]: nit 
```suggestion
`0.33.0`+ _only_ supports Karpenter v1beta1 APIs and will not work with existing Provisioner, AWSNodeTemplate or Machine alpha APIs. Do not upgrade to `0.36.0`+ without first [upgrading to `0.32.x`]({{<ref "#upgrading-to-0320" >}}). This version supports both the alpha and beta APIs, allowing you to migrate all of your existing APIs to beta APIs without experiencing downtime.
```

Comment (jonathan-innis) [reply website/content/en/preview/upgrading/upgrade-guide.md]: Updated, nice catch.

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Pull request merged**